lab/dp/problems/custi/main.cpp

```diff
@@ -0,0 +1,135 @@
+// Copyright 2020 Nicolae Mara-Ioana <maraioana9967@gmail.com>
+
+// Custi - Programare dinamica
+// Complexitate: O(n^2)
+
+// https://www.infoarena.ro/problema/custi
+
+#include <bits/stdc++.h>
+
+class Task {
+ public:
+    void solve() {
+        read_input();
+        get_result();
+        print_output();
+    }
+
+ private:
+    int n; // dimensiunea matricei
+    
+    std::vector<std::vector<int>> matrix; // matricea de input
+
+    // fiecare element solution[i] reprezinta numarul de custi
+    // de dimensiune maxima i x i ce se pot forma
+    // fiecare celula se foloseste o singura data ca si colt dreapta jos
+    // a unei custi, aceasta fiind cea de dimesniune maxima
+    std::vector<int> solution;
+
+    // citeste datele de intare din fisier si populeaza matricea
+    void read_input() { 
+        std::cin >> n;
+        matrix = std::vector<std::vector<int>>(n, std::vector<int>(n, 0));
+        solution = std::vector<int>(n + 1, 0);
```

Comment:
Does `solution = {n + 1, 0}` work here? Same at line 32.

Reply:
Tried doing this for matrix, but the following warning appears:
narrowing conversion of ‘(((Task*)this)->Task::n + 1)’ from ‘int’ to ‘std::vector<std::vector<int> >::size_type {aka long unsigned int}’ inside { } [-Wnarrowing]
         matrix = {n + 1, {n + 1, 0}};

Reply:
Maybe a few paranthesis? Or can you try using std::vector::std::vector<>(...)?

Reply:
Tried, that but it doesn't seem to work on Infoarena. I've already used std::vector::std::vector<>(...) in the first draft.